src/common/stack.rs

```diff
@@ -1,88 +1,419 @@
+use core::marker::PhantomData;
+use core::ptr;
+use core::usize;
+
 #[allow(unused_imports)]
 use alloc::prelude::*;
 
-#[cfg(feature = "std")]
-use std::error;
-use core::fmt;
+#[derive(Copy, Clone, Debug)]
+pub struct StackOverflow;
```

Comment:
Why remove the stack information from the error? I'm not saying that we necessarily need it, but I would like to know your justification.

Reply:
The interpreter converts `StackOverflow` to `TrapKind::StackOverflow` before passing it back to the user. Figured I'd start with a zero-length type to match `TrapKind::StackOverflow`. It would be easy to add extra fields if they are deemed useful.